spec/services/get_products_service_spec.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe GetProductsService do
+  subject(:service) { described_class.new(category_id) }
+
+  let(:category_id) { SecureRandom.uuid }
+  let(:all_category_id) { SecureRandom.uuid }
+  let(:parent_category_id) { SecureRandom.uuid }
+
+  let!(:all_category) { create(:category, id: all_category_id) }
+  let!(:first_parent_category) { create(:category, id: parent_category_id, parent: all_category) }
+  let!(:first_child_category) { create(:category, id: category_id, parent: first_parent_category) }
+  let!(:second_child_category) { create(:category, parent: first_parent_category) }
+
+  let!(:second_parent_category) { create(:category, parent: all_category) }
+
+  let!(:first_product) { create(:product, category: first_child_category) }
+  let!(:second_product) { create(:product, category: second_child_category) }
+
+  let!(:third_product) { create(:product, category: second_parent_category) }
+
+  describe '#validate' do
+    context 'when category_name does not exist in db' do
```

Comment:
check the context, u expect id but the context said name ?, please be more responsibility to ur code and double check it